src/server/scripts/Outland/zone_blades_edge_mountains.cpp

```diff
@@ -30,78 +30,178 @@ EndContentData */
 #include "SpellAuraEffects.h"
 
 // Ours
-enum deathsDoorWrapGate
+enum deathsdoorfell
 {
     SPELL_ARTILLERY_ON_THE_WRAP_GATE            = 39221,
-    SPELL_ANTI_DEMON_FLAME_THROWER              = 39222,
+    SPELL_IMP_AURA                              = 39227,
+    SPELL_HOUND_AURA                            = 39275,
+    SPELL_EXPLOSION                             = 30934,
+
+    NPC_DEATHS_DOOR_FEL_CANNON_TARGET_BUNNY     = 22495,
+    NPC_DEATHS_DOOR_FEL_CANNON                  = 22443,
+    NPC_EXPLOSION_BUNNY                         = 22502,
+    NPC_FEL_IMP                                 = 22474,
+    NPC_HOUND                                   = 22500,
     NPC_NORTH_GATE                              = 22471,
     NPC_SOUTH_GATE                              = 22472,
     NPC_NORTH_GATE_CREDIT                       = 22503,
     NPC_SOUTH_GATE_CREDIT                       = 22504,
 
+    GO_FIRE                                     = 185317,
+    GO_BIG_FIRE                                 = 185319,
+
+    EVENT_PARTY_TIMER                           = 1
 };
 
-class npc_deahts_door_wrap_gate : public CreatureScript
+class npc_deaths_door_fell_cannon_target_bunny : public CreatureScript
 {
 public:
-    npc_deahts_door_wrap_gate() : CreatureScript("npc_deahts_door_wrap_gate") { }
+    npc_deaths_door_fell_cannon_target_bunny() : CreatureScript("npc_deaths_door_fell_cannon_target_bunny") { }
 
-    CreatureAI* GetAI(Creature* creature) const
+    struct npc_deaths_door_fell_cannon_target_bunnyAI : public ScriptedAI
     {
-        return new npc_deahts_door_wrap_gateAI(creature);
-    }
+        npc_deaths_door_fell_cannon_target_bunnyAI(Creature* creature) : ScriptedAI(creature), PartyTime(false) { }
 
-    struct npc_deahts_door_wrap_gateAI : public ScriptedAI
-    {
-        npc_deahts_door_wrap_gateAI(Creature *c) : ScriptedAI(c)
+        EventMap events;
+        bool PartyTime;
+        uint64 PlayerGUID;
+        uint64 CannonGUID;
+        uint8 count;
+        
+        void Reset() override
         {
+            Initialize();
+            events.Reset();
         }
 
-        uint8 count;
-        uint32 timer;
-        void Reset() { count = 0; timer = 0;}
-        void SpellHit(Unit* caster, SpellInfo const* spellInfo)
+        void Initialize()
         {
-            if (spellInfo->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
-            {
-                Player* plr = caster->GetCharmerOrOwnerPlayerOrPlayerItself();
-                if (!plr)
-                    return;
+            PartyTime = false;
+            PlayerGUID = 0;
+            CannonGUID = 0;
+            count = 0;
+        }
 
-                timer = 1;
+        void SpellHit(Unit* caster, SpellInfo const* spell) override
+        {
+            if (spell->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
+            {
                 count++;
 
+                if (count)
```

Comment:
This should only be executed on the first shot. Otherwise multiple events are triggered, swarming the player with summoned adds.
```suggestion
                if (count == 1)
```